agent/structs/acl.go

```diff
@@ -244,6 +274,8 @@ func (t *ACLToken) SetHash(force bool) []byte {
 			hash.Write([]byte(link.ID))
 		}
 
+		// TODO(rb): put the expiration time here?
```

Comment:
Any non-immutable "content" fields should be involved with the overall hash. The IDs are immutable which is why they aren't here. The raft indices are metadata similar to the hash which is why they aren't incorporated. CreateTime is similarly immutable

While you are in here you may want to add a comment about what should go into the hash.

The Hash is really only used for replication to determine if a token has changed and should be updated locally.

Reply:
Unless I am mistaken you are considering the ExpirationTime immutable. Why not allow a token to be updated with a new expiration time?

Reply:
> Why not allow a token to be updated with a new expiration time?

Having tokens be renewable (rather than one-shot) starts down a path that @jefferai had suggested we avoid due to the cognitive overhead and operational complexity that Vault incurs from having a similar feature.

Reply:
It is more complex for sure. But without it you may get into the situation where for whatever reason you are unable to distribute a new token somewhere and the current token has an expiration. Time bombs like this that cannot be mitigated on the servers seems like a bad route to go down.

Reply:
For now this is probably fine but I think we may want to reconsider the ramifications of the decision.

Reply:
Thought it is being exposed as a first class feature out of a plumbing necessity, this is primarily being introduced as an implementation detail for the upcoming Login workflow.